### PR TITLE
feat(components): [input] add input formatter

### LINKED ISSUE #6652: [Feature Request] [Enhancement] [input] Suggest adding thousand formatting for el-input
Feature type: **`Enhancement`**

## Details

- Enhanced component name: `input`

## Description

It is a common requirement to enter numbers and format them into thousands. 

image

There have been several issues in element 2.0, but no response.

- [input-number support thousands of bits](https://github.com/ElemeFE/element/issues/15490)
- [display thousands separators formatting for Input Number](https://github.com/ElemeFE/element/issues/21181)

### Additional comments
If this feature passes the discussion, I am willing to develop it.

### PR DESCRIPTION
**Pull request by Serendipity96**
close #6652

add input formatter.

image

Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Review by YunYouJun — APPROVED** [review]

**Comment by YunYouJun** [comment]
@JeremyWuuuuu @sxzz Any other suggestions?

**Comment by sxzz** [comment]
I'll review it later

**Review by sxzz — APPROVED** [review]

File: packages/hooks/use-cursor/index.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import { type ShallowRef, ref } from 'vue'
```
Comment (sxzz) [inline packages/hooks/use-cursor/index.ts:1]:
```suggestion
import { ref } from 'vue'

import type { ShallowRef } from 'vue'
```

File: packages/components/input/src/input.vue
Comment (vinsonw) [inline packages/components/input/src/input.vue:1]: Line 48/49: What are the `formatter` and `parse` props on input element for? They're not native attributes for input element.

File: packages/components/input/src/input.vue [hunk 2]
```diff
@@ -45,6 +45,8 @@
           v-bind="attrs"
           :type="showPassword ? (passwordVisible ? 'text' : 'password') : type"
           :disabled="inputDisabled"
+          :formatter="formatter"
```
Comment (vinsonw) [inline packages/components/input/src/input.vue:48]: Line 48/49: What are the formatter and parse props on input element for? They're not native attributes for input element.

**Pull request merged**